alacritty_terminal/src/term/mod.rs

```diff
@@ -501,6 +501,57 @@ impl<T> Term<T> {
         }
     }
 
+    /// Go to next prompt farthest down in history.
+    pub fn goto_next_prompt(&mut self)
+    where
+        T: EventListener,
+    {
+        let point = if self.mode.contains(TermMode::VI) {
+            self.vi_mode_cursor.point
+        } else {
+            let display_offset = self.grid.display_offset() as i32;
+            // If we're outside of Vi mode use bottom most visible line.
+            Point::new(Line(self.screen_lines() as i32 - 1 - display_offset), Column(0))
+        };
+
+        let prompt_point = match self.next_prompt(point) {
+            Some(point) => point,
+            None => return,
+        };
+
+        if self.mode().contains(TermMode::VI) {
+            self.vi_goto_point(prompt_point);
+        } else {
+            let scroll = Scroll::Delta(point.line.0 - prompt_point.line.0);
+            self.scroll_display(scroll);
+        }
+    }
+
+    /// Go to previous prompt farthest up in history.
```

Comment:
We don't go to the previous prompt farthest up in history, that would mean we skip over as many prompts as possible. We go to the previous prompt in history, so the opposite of farthest. Could say closest/nearest up in history but I feel like that's unnecessary.